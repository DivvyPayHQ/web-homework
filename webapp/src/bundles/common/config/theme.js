import * as COLORS from './colors'

export const THEME_TYPES = {
  LIGHT: 'LIGHT',
  DARK: 'DARK'
}

const LIGHT = {
  type: THEME_TYPES.LIGHT,
  background: COLORS.WHITE,
  secondary: COLORS.WHITE_SECONDARY,
  color: COLORS.BLACK,
  accent: COLORS.LIGHT_ACCENT,
  highlight: COLORS.PURPLE
}

const DARK = {
  type: THEME_TYPES.DARK,
  background: COLORS.BLACK,
  secondary: COLORS.BLACK_SECONDARY,
  color: COLORS.WHITE,
  accent: COLORS.DARK_ACCENT,
  highlight: COLORS.PURPLE
}

export function getTheme (themeType) {
  if (themeType === THEME_TYPES.LIGHT) {
    return LIGHT
  }
  return DARK
}
