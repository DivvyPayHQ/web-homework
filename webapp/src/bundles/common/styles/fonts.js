import { css } from '@emotion/core'

export const defaultFont = css`
    font-family: "Open Sans",serif;
`
