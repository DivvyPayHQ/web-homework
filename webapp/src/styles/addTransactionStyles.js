import { css } from '@emotion/react'

export const addTransactionStyle = css`
        width: 35%;
        margin: 0 auto;
        display: flex;
        flex-direction: column;

    .addTransactionHeader{
        color: #487ECE;
        text-transform: uppercase;
        font-size: 1.2rem;
        margin-right: 35%;
        margin-top: 2%;
    }

    .addTransactionForm {
        width: 80%;
        display: flex;
        flex-direction: column;
        justify-content: space-evenly;
    }

    .addTransactionInput {
        background-color: #dedfe0;
        height: 3vh;
        margin-bottom: 1rem;
        font-weight: bold;
    }

    label {
        text-transform: uppercase;
        font-size: .8rem;
    }
    
    .addTransactionRadio {
        position: relative;
        right: 7.9rem;
        bottom: 1.2rem;
    }

    button {
        width: 15%;
        height: 4vh;
        border-radius: 2rem;
        background-color: black;
        color: white;
        text-transform: uppercase;
        font-size: .6rem;
        font-weight: bold;
    }
`
