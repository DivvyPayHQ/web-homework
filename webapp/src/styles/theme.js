export const theme = {
  colors: {
    primaryBackground: 'white',
    primary: '#0062B8',
    success: '#18891A',
    info: '#1564DB',
    warning: '#F9A109',
    danger: '#D62620',
    black: '#000'
  },
  typography: {
    fontFamily: [
      'Montserrat',
      'sans-serif'
    ].join(',')
  },
  fontFamily: 'Montserrat, sans-serif'
}
