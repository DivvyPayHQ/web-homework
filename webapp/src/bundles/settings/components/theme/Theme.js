import React from 'react'
import { shape, string, func } from 'prop-types'
import { connect } from 'react-redux'
import Switch from 'Components/switch/Switch'
import { setTheme } from 'Actions/setTheme'
import { THEME_TYPES } from 'Config/theme'
import { css } from '@emotion/core'

function Theme ({ theme, setTheme }) {
  const darkMode = theme.type === THEME_TYPES.DARK
  return (
    <div css={containerStyles}>
      <p css={labelStyles} style={{ color: theme.color }}>Dark Mode</p>
      <Switch
        on={darkMode}
        onClick={() => darkMode ? setTheme(THEME_TYPES.LIGHT) : setTheme(THEME_TYPES.DARK)}
        theme={theme}
      />
    </div>
  )
}

const containerStyles = css`
  display: flex;
  align-items: center;
`

const labelStyles = css`
  margin-right: 20px;
`

function mapDispatchToProps (dispatch) {
  return {
    setTheme: (themeType) => { dispatch(setTheme(themeType)) }
  }
}

export default connect(null, mapDispatchToProps)(Theme)

Theme.propTypes = {
  theme: shape({
    type: string,
    background: string,
    secondary: string,
    color: string,
    accent: string,
    highlight: string
  }),
  setTheme: func.isRequired
}
