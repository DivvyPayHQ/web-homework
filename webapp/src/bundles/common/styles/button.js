import { css } from '@emotion/core'
import * as COLORS from 'Config/colors'
import { defaultFont } from 'Config/fonts'

export const baseButton = css`
  width: 100%;
  border: none;
  border-radius: 5px;
  padding: 10px;
  font-family: ${defaultFont};
  font-size: 13px;
  font-weight: 700;
  color: ${COLORS.WHITE};
  text-transform: uppercase;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 30px;

  &:focus {
    outline: none;
  }

  &:hover {
    opacity: 0.8;
  }
`
