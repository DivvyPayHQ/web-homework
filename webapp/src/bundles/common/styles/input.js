import { css } from '@emotion/core'
import { defaultFont } from 'Config/fonts'

export const baseInput = css`
  border-radius: 5px;
  border: solid 1px;
  box-sizing: border-box;
  font-family: ${defaultFont};
  padding: 5px;
  font-size: 14px;
  min-width: 0;
  &:focus {
    outline: none;
  }
`
