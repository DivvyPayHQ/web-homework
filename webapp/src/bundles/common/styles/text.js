import { css } from '@emotion/core'
import { defaultFont } from '../config/fonts'

export const baseLabel = css`
  font-family: ${defaultFont};
  font-weight: 600;
  font-size: 15px;
  width: 100%;
  text-align: left;
  text-transform: capitalize;
  margin: 0 0 5px;
`

export const baseValue = css`
  font-family: ${defaultFont};
  font-size: 15px;
  font-weight: 300;
  margin: 0;
`
