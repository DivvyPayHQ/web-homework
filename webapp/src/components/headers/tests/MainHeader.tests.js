import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from '@emotion/react'
import { MainHeader } from "../"
import { theme } from '../../../styles/theme'

describe("MainHeader", () => {
  const component = (
    <ThemeProvider theme={theme}>
      <MainHeader />
    </ThemeProvider>
  )

  it("renders successfully and matches snapshot", () => {
    const mounted = renderer.create(component).toJSON()
    expect(mounted).toMatchSnapshot()
  })
})
