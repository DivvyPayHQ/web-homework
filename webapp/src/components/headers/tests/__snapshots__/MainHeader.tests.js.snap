// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader renders successfully and matches snapshot 1`] = `
.emotion-2 {
  width: 100%;
  padding: .5em 0;
}

.emotion-0 {
  color: #000;
  margin: 0;
  font-family: Montserrat,sans-serif;
  letter-spacing: .075em;
  text-transform: capitalize;
}

<section
  className="emotion-2 emotion-3"
>
  <h2
    className="emotion-0 emotion-1"
  />
</section>
`;
